Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift

```diff
@@ -206,6 +218,32 @@ open class ChatMessageListView: UITableView, Customizable, ComponentsProvider {
         reloadData()
         scrollToMostRecentMessage()
     }
+
+    /// Adjusts the content inset so that messages are inserted at the top when there are few messages.
+    /// This is will be executed if the `Components.shouldMessagesStartAtTheTop` is enabled.
+    internal func adjustContentInsetToPositionMessagesAtTheTop() {
+        guard components.shouldMessagesStartAtTheTop else {
+            return
+        }
+
+        // If the height of message list is more than the content height
+        // then adjust the content inset so that it fills the remaining height
+        // otherwise do not set any content inset.
+        let contentSizeHeight = contentSize.height
+        let messageListHeight = frame.height
+        let newContentInset = messageListHeight - contentSizeHeight
+        if newContentInset > 0 {
+            contentInset.top = newContentInset
+            showsVerticalScrollIndicator = false
+            requiresContentInsetReset = true
+            // In case  we already removed the content inset, there's
+            // no need to do it every time.
+        } else if requiresContentInsetReset {
+            requiresContentInsetReset = false
+            contentInset.top = 0
+            showsVerticalScrollIndicator = true
+        }
```

Comment:
nit: Just for readability purposes, can we add the `else` with `no-op` and probably `showsVerticalScrollIndicator = false`?

Reply:
Not sure what you mean. Can you clarify?

Reply:
```suggestion
} else {
 // No-op
}
```

Reply:
It improves readability and explicitly defines all possible paths of the flow. It's a nice to have if we want to ensure that readability is maintained to high standards but I'm happy to leave it for another PR.